src/error.ts

```diff
@@ -1237,11 +1242,13 @@ export function needsRetryableWriteLabel(error: Error, maxWireVersion: number):
   }
 
   if (error instanceof MongoWriteConcernError) {
-    return RETRYABLE_WRITE_ERROR_CODES.has(error.result?.code ?? error.code ?? 0);
+    return serverType === 'Mongos' && maxWireVersion < 9
```

Comment:
I don't think this is right still - the spec says that for <4.4 mongoses, we _do_ check the top level code, we just don't check writeConcernError.code.  As written, I think that this will always return false if a `writeConcernError` is returned with a top-level retryable error code, when instead we should be hitting the if-statement on line 1250.

Right?

Reply:
Right, I'm saying that I think it's supposed to reach line 1250 but for a pre-4.4 mongos writeConcernError, we will always return `false` here.

Reply:
Bookkeeping, we are now checking only the top-level code (not nested code) for 4.2 sharded clusters